### ISSUE TITLE
flux calculations when using set_full_model

### ISSUE THREAD

**Issue by anetasie** [sherpa/sherpa#966]
`set_full_model` gives the flexibility in setting the model expression and one can calculate the flux for different model components. The updated `sample_energy_flux` handles this case. 
[bkg.example.tar.gz](https://github.com/sherpa/sherpa/files/5344013/bkg.example.tar.gz)

```
sherpa In [17]: sample_energy_flux(0.5,7.,model=pow1)
Out[17]: array([[2.18501706e-12, 5.85561312e-01, 3.31314089e+00, 7.43932636e-04]])
````

However,  `sample_flux` fails. We could improve the current error msg saying `You should use sample_energy_flux or sample_photon_flux instead`

```
sherpa In [20]: sample_flux(pow1,0.5,7.)
IdentifierErr: Convolved model
'(apply_rmf(apply_arf((50454.152030948 * (xsphabs.abs1 * powlaw1d.pow1)))) + (0.06576568179195134 * apply_rmf(apply_arf((50454.152030948 * (const1d.c1 * (((((((powlaw1d.bkg_pow1 + powlaw1d.bkg_pow2) + gauss1d.bkg_g1) + gauss1d.bkg_g2) + gauss1d.bkg_g3) + gauss1d.bkg_g4) + gauss1d.bkg_g5) + gauss1d.bkg_g6)))))))'
 is set for dataset 1. You should use get_model instead.
```

**Comment by DougBurke** [comment]
Well, I was working on fixing sample_flux in #754 but that got delayed by work that I needed to get done before being able to fix sample_flux. I'm still waiting for SDS input on #866 before I can work on sample_flux.

**Comment by DougBurke** [comment]
@anetasie - can  you try this with master and/or #1106 to see if you still see this error (you probably will maybe we're lucky)

**Comment by anetasie** [comment]
@DougBurke No luck! I still get the same error msg with both #1106 and main.

IdentifierErr: Convolved model
'(apply_rmf(apply_arf((50454.152030948 * powlaw1d.pow1))) + (0.06576568179195134 * apply_rmf(apply_arf((50454.152030948 * (const1d.c1 * (((((((powlaw1d.bkg_pow1 + powlaw1d.bkg_pow2) + gauss1d.bkg_g1) + gauss1d.bkg_g2) + gauss1d.bkg_g3) + gauss1d.bkg_g4) + gauss1d.bkg_g5) + gauss1d.bkg_g6)))))))'
 is set for dataset 1. You should use get_model instead.

**Comment by DougBurke** [comment]
I claim that this is now addressed in #1106:

```python
In [3]: sample_flux(pow1, 0.5, 2)
---------------------------------------------------------------------------
IdentifierErr                             Traceback (most recent call last)
~/sherpa/bugs/issue966/bkg_example.py in <module>
----> 1 sample_flux(pow1, 0.5, 2)

<string> in sample_flux(modelcomponent, lo, hi, id, num, scales, correlated, numcores, bkg_id, Xrays, confidence)

~/sherpa/sherpa-master/sherpa/astro/ui/utils.py in sample_flux(self, modelcomponent, lo, hi, id, num, scales, correlated, numcores, bkg_id, Xrays, confidence)
  13561                 self.get_source(id)
  13562             except IdentifierErr:
> 13563                 raise IdentifierErr('Please use calc_energy_flux as set_full_model was used') from None
  13564
  13565         correlated = sherpa.utils.bool_cast(correlated)

IdentifierErr: Please use calc_energy_flux as set_full_model was used
```

The error message isn't great but I have no energy (or desire) to improve it.
